Sources/SwiftCentrifuge/Client.swift

```diff
@@ -1035,6 +997,10 @@ fileprivate extension CentrifugeClient {
                 )
             }
         }
+
+        if self.state == .connecting {
+            self.scheduleReconnect()
```

Comment:
Usually we only want reconnect scheduled only after connection was really closed. That's why it was inside onDisconnect transport callback. The reason was to avoid connection leaks - `processDisconnect ` may be called several time if I remember right – so need to make sure we are not creating extra conns here.

Ideally we better continue keeping `scheduleReconnect` in `onDisconnect` handler. Not sure whether it's possible though given reasons that lead to this refactoring here.

Reply:
Moved `scheduleReconnect` back to `onDisconnect `.